core/src/main/scala/org/apache/spark/ExecutorAllocationClient.scala

```diff
@@ -59,6 +59,19 @@ private[spark] trait ExecutorAllocationClient {
   def killExecutors(executorIds: Seq[String]): Seq[String]
 
   /**
+   * Request that the cluster manager try harder to kill the specified executors,
+   * and maybe replace them.
+   * @return whether the request is acknowledged by the cluster manager.
+   */
+  def killExecutors(executorIds: Seq[String], replace: Boolean, force: Boolean): Seq[String]
```

Comment:
Let's go with default values for now. I agree that it is good to collapse these into one function.